src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -1354,14 +1355,39 @@ protected void calculateLengthFromGeometry () {
                 e.setFloatingVehicleDropoffSuitability(getFloatingVehicleDropoffSuitability());
             }
         } else {
-            if (((TemporarySplitterVertex) v).isEndVertex()) {
-                e1 = new TemporaryPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name);
-                e1.setNoThruTraffic(this.isNoThruTraffic());
-                e1.setStreetClass(this.getStreetClass());
+            if (createSemiPermanentEdges) {
+                e1 = new SemiPermanentPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name);
+                e2 = new SemiPermanentPartialStreetEdge(this, v, (StreetVertex) tov, geoms.second, name);
+
+                for (StreetEdge e : new StreetEdge[] { e1, e2 }) {
+                    e.setBicycleSafetyFactor(getBicycleSafetyFactor());
+                    e.setHasBogusName(hasBogusName());
+                    e.setStairs(isStairs());
+                    e.setWheelchairAccessible(isWheelchairAccessible());
+                    e.setBack(isBack());
+                    e.setNoThruTraffic(isNoThruTraffic());
+                    e.setStreetClass(getStreetClass());
+                }
             } else {
-                e2 = new TemporaryPartialStreetEdge(this, v, (StreetVertex) tov, geoms.second, name);
-                e2.setNoThruTraffic(this.isNoThruTraffic());
-                e2.setStreetClass(this.getStreetClass());
+                if (((TemporarySplitterVertex) v).isEndVertex()) {
+                    // There is no need to split a SemiPermanentPartialStreetEdge when the to vertex is a
+                    // SemiPermanentSplitterVertex. The original edge that the SemiPermanentPartialStreetEdge was split
+                    // from will also be split and we'd end up with basically 2 identical TemporaryPartialStreetEdges.
+                    if (!(this instanceof SemiPermanentPartialStreetEdge && tov instanceof SemiPermanentSplitterVertex)) {
+                        e1 = new TemporaryPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name, 0);
+                        e1.setNoThruTraffic(this.isNoThruTraffic());
+                        e1.setStreetClass(this.getStreetClass());
+                    }
+                } else {
+                    // There is no need to split a SemiPermanentPartialStreetEdge when the from vertex is a
+                    // SemiPermanentSplitterVertex. The original edge that the SemiPermanentPartialStreetEdge was split
+                    // from will also be split and we'd end up with basically 2 identical TemporaryPartialStreetEdges.
+                    if (!(this instanceof SemiPermanentPartialStreetEdge && fromv instanceof SemiPermanentSplitterVertex)) {
```

Comment:
Since this condition is checked for twice, I think it could make since to move it up a level and only check once.
```
boolean splitOnEnd = ((TemporarySplitterVertex) v).isEndVertex();
StreetVertex outerVertex = splitOnEnd ? (StreetVertex) tov : (StreetVertex) fromv;
if (this instanceof SemiPermanentPartialStreetEdge && outerVertex instanceof SemiPermanentSplitterVertex) {
    // There is no need to split a SemiPermanentPartialStreetEdge when the outer vertex is a
    // SemiPermanentSplitterVertex. The original edge that the SemiPermanentPartialStreetEdge was split
    // from will also be split and we'd end up with basically 2 identical TemporaryPartialStreetEdges.
    return;
} else {
    if (splitOnEnd) { ... }
   else { ... }
}
```

Reply:
ok, will refactor, but with different variable names. It's somewhat hard to wrap my brain around how to describe what the `outerVertex` is here.